fala/apps/adviser/tests/page_objects.py

```diff
@@ -0,0 +1,75 @@
+class FalaPage(object):
+    def __init__(self, page):
+        self._page = page
+
+    @property
+    def h1(self):
+        return self._page.locator("h1")
+
+    @property
+    def language_dropdown(self):
+        return self._page.locator('select[class="goog-te-combo"]')
+
+    def item_from_text(self, text):
+        return self._page.get_by_text(text)
+
+    def select_specific_page(self, page_no):
+        return self._page.locator(f'a.govuk-pagination__link:has-text(" {page_no} ")')
+
+    def change_search(self):
+        self.item_from_text("Change Search").click()
+        return SearchPage(self._page)
+
+
+class ResultsPage(FalaPage):
+    @property
+    def next_link(self):
+        return self._page.locator('span.govuk-pagination__link-title:has-text(" Next")')
+
+    @property
+    def pagination_link_title(self):
+        return self._page.locator("span.govuk-pagination__link-title")
+
+    @property
+    def previous_link(self):
+        return self._page.locator('span.govuk-pagination__link-title:has-text(" Previous")')
+
+    @property
+    def change_search_grey_box(self):
+        return self._page.locator("li.govuk-body.notranslate")
+
+    def listitem_for(self, postcode):
```

Comment:
at the minute this just takes a postcode but you could also refactor to work for organisation and categories too.

Reply:
`listitem_for` is specific for one test scenario, I don't see the value in making it more adaptable.  We have `visit_results_page` & `visit_results_page_with_full_search` which would cover majority of the journeys we want to test and it's very readable as to what you are testing when searching.


I could make it more adaptable with doing something like this below but I don't think it's worth it. 

> def listitem_for(self, text_type, text):
>     return self._page.get_by_role("listitem").filter(has_text=f"{text_type}: {text}")